shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -434,7 +437,90 @@ private int getPointerDeviceTypeForToolType(int toolType) {
         }
     }
 
-    private void addPointerForIndex(MotionEvent event, int pointerIndex, int pointerChange,
+    private List<PointerAction> expandActions(MotionEvent event, int pointerIndex, int pointerChange,
+    int batched) {
+        long device = event.getPointerId(pointerIndex);
+        // the android touch events do not always follow clear
+        // guideline. We normalize the events so that the framework
+        // does not have to deal with android specific issue.
+        double dx = event.getX(pointerIndex);
+        double dy = event.getY(pointerIndex);
+        PointerState state;
+        List<PointerAction> requiredAction = new ArrayList<PointerAction>();
+        switch (pointerChange) {
+            case kPointerChangeCancel:
```

Comment:
I would order the cases roughly by the expected life cycle, e.g. start with `add`